programs/stake/src/rewards.rs

```diff
@@ -185,11 +185,17 @@ fn calculate_stake_rewards(
 
     let rewards = points
         .checked_mul(u128::from(point_value.rewards))
-        .unwrap()
-        .checked_div(point_value.points)
-        .unwrap();
+        .expect(&format!(
+            "Rewards should fit within u128, inputs were {} and {}",
+            points, point_value.rewards
+        ));
 
-    let rewards = u64::try_from(rewards).unwrap();
+    // The unwrap is safe, as points_value.points is guaranteed to be non zero above.
+    // Expect is verifying that the result fits in u64
+    let rewards = u64::try_from(rewards.checked_div(point_value.points).unwrap()).expect(&format!(
+        "Rewards should fit within u64, inputs were {}, {} and {}",
+        points, point_value.rewards, point_value.points
+    ));
```

Comment:
Sorry for only doing this now, but I ran a bench on this PR, and just adding the string formatting has a surprising negative performance impact, causing this function to go from ~5.28ns to ~7ns.

Since rewards payouts are a huge bottleneck, I'm hesitant to do anything that will negatively impact performance. I also tried changing to `unwrap_or_else` to panic lazily, but it made no difference. If we use static strings, however, the performance is exactly the same. What do you think about using static strings instead? That would be just "Rewards should fit within u128" and "Rewards should fit within u64"

Reply:
Yeah for sure! Here's my bench -- it requires making a few extra structs public to make it work, but I hope it provides a good start:

```
#![feature(test)]

extern crate test;
use {
    solana_stake_program::{points::{PointValue, null_tracer}, rewards::calculate_stake_rewards, stake_state::new_stake},
    solana_vote_program::vote_state::VoteState,
    solana_sdk::{
        pubkey::Pubkey,
        stake_history::StakeHistory,
    },
    test::Bencher,
};


#[bench]
fn calculate_rewards_tests(bencher: &mut Bencher) {
    let stake = 1_000_000_000;
    let rewards = 20_000_000_000;
    let credits = 30_000_000_000;

    let mut vote_state = VoteState::default();

    let stake = new_stake(stake, &Pubkey::default(), &vote_state, 100_000_000_000);

    vote_state.increment_credits(0, credits);

    bencher.iter(|| {
        calculate_stake_rewards(
            0,
            &stake,
            &PointValue {
                rewards: rewards,
                points: 1,
            },
            &vote_state,
            &StakeHistory::default(),
            null_tracer(),
            None,
        );
    });
}
```

Reply:
Yup, I got it to work locally. Seeing similar results that make any kind of logging a no go. 

One interesting/very surprising result: 
'''
    let rewards = points
        .checked_mul(u128::from(point_value.rewards))
        .expect(&format!(
            "Rewards should fit within u128, inputs were {}", 3 as u32
        ));
'''
test rewards::tests::bench_add_two ... bench:          38.25 ns/iter (+/- 0.54)

'''
    let rewards = points
        .checked_mul(u128::from(point_value.rewards))
        .expect(&format!(
            "Rewards should fit within u128, inputs were {}", 3
        ));
'''
test rewards::tests::bench_add_two ... bench:          22.27 ns/iter (+/- 0.67)

Reply:
On further testing, even having the expects statement is causing a performance delta.

Reply:
On my testing, when I did `expect("static string")` it recovered the original performance, which makes sense since there's no extra work required. If you wanted to do that, I'd be down!

Reply:
I tested that and still saw significant performance delta. 
Base code: 
test rewards::tests::bench_pow ... bench:           8.78 ns/iter (+/- 0.32)

With expects using static string:
test rewards::tests::bench_pow ... bench:          23.16 ns/iter (+/- 0.43)